src/h2/utilities.py

```diff
@@ -515,14 +514,14 @@ def utf8_encode_headers(headers):
     tuples that preserve the original type of the header tuple for tuple and
     any ``HeaderTuple``.
     """
-    return [
-        (
-            header.__class__(_to_bytes(header[0]), _to_bytes(header[1]))
-            if isinstance(header, HeaderTuple)
-            else (_to_bytes(header[0]), _to_bytes(header[1]))
-        )
-        for header in headers
-    ]
+    encoded_headers = []
+    for header in headers:
+        h = (_to_bytes(header[0]), _to_bytes(header[1]))
+        if isinstance(header, HeaderTuple):
+            encoded_headers.append(header.__class__(h[0], h[1]))
+        else:
+            encoded_headers.append(h)
+    return encoded_headers
```

Comment:
This is again, quite less efficient compared to using a list-comprehension directly.